stytch/b2b/b2bstytchapi/b2bstytchapi.go

```diff
@@ -47,6 +49,26 @@ func WithBaseURI(uri string) Option {
 	return func(api *API) { api.client.Config.BaseURI = config.BaseURI(uri) }
 }
 
+// NewClient returns a Stytch API client that uses the provided credentials.
+//
+// It detects the environment from the given projectID. You are still free to pass WithBaseURI as an option if you wish
+// to override this behavior, but the intention is to provide a simpler interface for creating a client since it's
+// extremely rare that developers would want to use something other than the detected environment.
+func NewClient(projectID string, secret string, opts ...Option) (*API, error) {
+	var detectedEnv config.Env
+	if strings.HasPrefix(projectID, "project-live-") {
+		detectedEnv = config.EnvLive
+	} else {
+		detectedEnv = config.EnvTest
+	}
+
+	return NewAPIClient(detectedEnv, projectID, secret, opts...)
+}
+
+// NewAPIClient returns a Stytch API client that uses the provided credentials.
+//
+// It is highly recommended to use NewClient instead of this function since it will automatically detect the correct
+// Stytch environment from the provided projectID. This function will be deprecated in a future MAJOR release.
```

Comment:
Based on this comment, it sounds like the plan is to leave this usable in v8, deprecate it in v9, and remove it in v10. 

I think we can move a little faster than that if you want to. We can deprecate it in a future minor version bump and then completely remove it in v9. I don't see any cases where someone would want `NewAPIClient(env, ...)` over `NewClient(..., WithBaseURI(override))` now that that exists.